api/resource_groups_aws.go

```diff
@@ -0,0 +1,94 @@
+//
+// Author:: Darren Murray (<darren.murray@lacework.net>)
+// Copyright:: Copyright 2021, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+import (
+	"encoding/json"
+)
+
+// GetAwsResourceGroup gets a single Aws ResourceGroup matching the
+// provided resource guid
+func (svc *ResourceGroupsService) GetAwsResourceGroup(guid string) (
+	response AwsResourceGroupResponse,
+	err error,
+) {
+	var rawResponse ResourceGroupResponse
+	err = svc.get(guid, &rawResponse)
+	if err != nil {
+		return AwsResourceGroupResponse{}, err
+	}
+
+	return setAwsResponse(rawResponse)
+}
+
+// UpdateAwsResourceGroup updates a single Aws ResourceGroup on the Lacework Server
+func (svc *ResourceGroupsService) UpdateAwsResourceGroup(data ResourceGroup) (
+	response AwsResourceGroupResponse,
+	err error,
+) {
+	var rawResponse ResourceGroupResponse
+	err = svc.update(data.ID(), data, &rawResponse)
+	if err != nil {
+		return AwsResourceGroupResponse{}, err
+	}
+
+	return setAwsResponse(rawResponse)
+}
+
+func setAwsResponse(response ResourceGroupResponse) (aws AwsResourceGroupResponse, err error) {
+	var props AwsResourceGroupProps
+	aws = AwsResourceGroupResponse{
+		Data: AwsResourceGroupData{
+			Guid:         response.Data.Guid,
+			IsDefault:    response.Data.IsDefault,
+			ResourceGuid: response.Data.ResourceGuid,
+			Name:         response.Data.Name,
+			Type:         response.Data.Type,
+			Enabled:      response.Data.Enabled,
+		},
+	}
+
+	err = json.Unmarshal([]byte(response.Data.Props.(string)), &props)
```

Comment:
knowing our APIs, if the `props` field changes from string to something else, this code will panic
trying to cast it, could we instead "try to cast" and if it does not work, return an error?
```suggestion
	propsString, ok := response.Data.Props.(string)
	if !ok {
		err = errors.New("unable to cast props field from API response")
		return
	}

	err = json.Unmarshal([]byte(propsString), &props)
```